ArduPlane/Attitude.cpp

```diff
@@ -5,31 +5,76 @@
   to PIDs to change the scaling of the PID with speed. At high speed
   we move the surfaces less, and at low speeds we move them more.
  */
-float Plane::calc_speed_scaler(void)
+Vector3f Plane::calc_rpy_speed_scaler(void)
 {
-    float aspeed, speed_scaler;
+    Vector3f speed_scaler;
+    float aspeed;
     if (ahrs.airspeed_estimate(aspeed)) {
         if (aspeed > auto_state.highest_airspeed && hal.util->get_soft_armed()) {
             auto_state.highest_airspeed = aspeed;
         }
+
+        Vector3f aspeed_rpy;
+        if (is_positive(g2.prop_disc_loading)) {
+            // Calculate prop-wash exit velocity using momentum disc theory
+            const float rho = SSL_AIR_DENSITY * plane.barometer.get_air_density_ratio();
+            // Estimate propeller disc loading in N/m^2 assuming drag is negligible and X axis 
+            // specific force is mostly due to propeller thrust. This assumption will underestimate
+            // propeller disc loading at higher speeds but will be accurate enough at low speeds
+            // which is when propwash effects dominate.
+            const float disc_loading = MAX(g2.prop_disc_loading * ahrs.get_accel().x, 0.0f);
+            // Solve the quadratic for propeller inflow factor a^2 + a = T / (2 * rho * S * V^2)
+            // a = inflow factor
```

Comment:
How are you defining this, ```a = (V - V_disk) / V``` ? In which case I would have expected the LHS of the equation to be ```a - a^2```

Reply:
Vs = Vo * (1 + 2a) where Vs is velocity of slipstream after flow is fully contracted, Vo is true airspeed of the aircraft, a is inflow factor

Formula from Aerodynamics for Engineering Students, Houghton and Carruthers, Third Edition, p469